src/main/scala/ch/openolitor/core/proxy/ProxyWorker.scala

```diff
@@ -77,9 +79,21 @@ class ProxyWorker(val serverConnection: ActorRef, val routeMap: Map[String, Mand
       log.error(s"Unmatched request: $x")
   }
 
+  val unsupported: Receive = {
+    case x =>
+      log.debug(s"Unsupported request state: $x")
+  }
+
   val requestContextHandshake: Receive = {
     case ctx: RequestContext =>
-      handshaking(ctx.request)
+      Try(handshaking(ctx.request)) match {
+        case Success(result) => result
+        case Failure(error) => {
+          log.error(s"Handshake failure: $error")
+          serverConnection ! HttpResponse(400)
```

Comment:
Just a question: Is BadRequest the correct answer? If it was a client error, maybe `403` would make more sense.
And if it's intended as a server side error maybe `503`.

Reply:
Or `426`